### PR TITLE
Add stop on fail flag to Experiment

### PR DESCRIPTION
**Pull request by nmatthews-asapp**
This PR adds a flag `stop_on_failure` to experiment that indicates whether a failed trial should terminate the entire experiment

### CODE REVIEW

File: flambe/experiment/experiment.py [hunk 1]
```diff
@@ -342,10 +344,17 @@ def trial_name_creator(trial):
...
 
+                any_error = False
                 for t in trials:
                     if t.status == t.ERROR:
                         logger.error(f"{t} ended with ERROR status.")
                         success[block_id] = False
+                        any_error = True
+                if any_error and self.stop_on_failure:
+                    self.teardown()
+                    logger.error(f"Stopping experiment at block '{block_id}' because there was an "
```
Comment (iitzco-asapp) [inline flambe/experiment/experiment.py]: we could add colors here. `cl.RE` for red one.

Comment (iitzco-asapp) [reply flambe/experiment/experiment.py]: Check this line https://github.com/asappresearch/flambe/pull/106/files#diff-7450e432b0e35241531af760c8a6d4eaR410

for reference

**Comment by iitzco-asapp** [comment]
Also I think we need a little more work on the progress state so that the report site shows this new state.

File: flambe/experiment/experiment.py (hunk 1 above)
Comment (nmatthews-asapp) [reply flambe/experiment/experiment.py]: `self.teardown()` updates the status I think this is okay

**Comment by jeremyasapp** [comment]
@iitzco-asapp do you want to add that in or should we go ahead and merge as is?

**Pull request merged**